gobblin-runtime/src/main/java/gobblin/runtime/AbstractJobLauncher.java

```diff
@@ -280,31 +281,44 @@ public void apply(JobListener jobListener, JobContext jobContext)
   /**
    * This predicate checks if a work unit should be skipped. If yes, then it will removed
    * from the list of workUnits and it's state will be saved.
+   *
+   * If complement is set to true, then the filter will be reversed
+   *
+   * If a MultiWorkUnit is marked as skipped all WorkUnits contained in it will be skipped.
+   * If a MultiWorkUnit contains some skipped WorkUnits, only those will be skipped.
    */
   @RequiredArgsConstructor
+  @AllArgsConstructor
   private static class SkippedWorkUnitsFilter implements Predicate<WorkUnit> {
     private final JobState jobState;
+    private boolean complement = false;
 
     @Override
     public boolean apply(WorkUnit workUnit) {
       if (workUnit instanceof MultiWorkUnit) {
-        Preconditions.checkArgument(!workUnit.contains(ConfigurationKeys.WORK_UNIT_SKIP_KEY),
-            "Error: MultiWorkUnit cannot be skipped");
-        for (WorkUnit wu : ((MultiWorkUnit) workUnit).getWorkUnits()) {
-          Preconditions.checkArgument(!wu.contains(ConfigurationKeys.WORK_UNIT_SKIP_KEY),
-              "Error: MultiWorkUnit cannot contain skipped WorkUnit");
+        WorkUnitStream wus = ((MultiWorkUnit) workUnit).getWorkUnitsStream();
```

Comment:
Is it necessary to use a `WorkUnitStream` for this? I think you could just do
```java
workUnit.getWorkUnits.stream().filter(new SkippedWorkUnitsFilter(jobState)).collect();
```
or using guava
```java
Iterables.filter(workUnit.getWorkUnits, Predicates.not(new SkippedWorkUnitsFilter(jobState)));
```